src/libs/ajax/WorkspaceDataService.ts

```diff
@@ -0,0 +1,25 @@
+import _ from 'lodash/fp'
+import { authOpts, fetchWDS, jsonBody } from 'src/libs/ajax/ajax-common'
+import { SearchRequest } from 'src/libs/ajax/data-table-providers/WDSDataTableProvider'
+
+
+export const WorkspaceDataService = signal => ({
+  getSchema: async (instanceId: string) => {
+    const res = await fetchWDS(`${instanceId}/types/v0.2`, _.merge(authOpts(), { signal }))
+    return res.json()
+  },
+  getRecords: async (instanceId: string, recordType: string, parameters: SearchRequest) => {
+    const res = await fetchWDS(`${instanceId}/search/v0.2/${recordType}`,
+      _.mergeAll([authOpts(), jsonBody(parameters), { signal, method: 'POST' }]))
+    return res.json()
+  },
+  deleteTable: async (instanceId: string, recordType: string) => {
+    const res = await fetchWDS(`${instanceId}/types/v0.2/${recordType}`,
+      _.mergeAll([authOpts(), { signal, method: 'DELETE' }]))
+    return res
+  },
+  downloadTsv: async (instanceId: string, recordType: string) => {
+    const res = await fetchWDS(`${instanceId}/tsv/v0.2/${recordType}`, _.merge(authOpts(), { signal }))
+    return res
+  }
```

Comment:
To make methods more consistent in returning response data vs the Response object, could this function return the response data as a blob instead of the Response itself?

```suggestion
  downloadTsv: async (instanceId: string, recordType: string): Promise<Blob> => {
    const res = await fetchWDS(`${instanceId}/tsv/v0.2/${recordType}`, _.merge(authOpts(), { signal }))
    const blob = await res.blob()
    return blob
  }
```